Added in support for preferRoot

mattdelliott opened the pull request:
This adds in the ability to pass an options object to the function. The two supported are optional `callback` or `preferRoot`

Also fixed an issue where it was looking at the destination filepath (installDirectory + appname.app) instead of just installDirectory for write permissions.

Merge if you find useful.

mattdelliott committed:
Updated to accept options and added support for preferRoot.

mattdelliott committed:
Bumped version to 0.0.6

mattdelliott committed:
Looking at installDirectory for permissions instead of app destination path

tommoor commented:
I agree that this behavior needs changing however I don't think an option at compile time is the right solution - you're basically saying if the application author uses the `preferRoot` option then never put the app in the users `~/Applications` directory as it stands…

This isn't the app author's choice to make though, on managed systems sometimes the users Applications directory is the *only* place that they will have permission to write to which is why this logic exists.

mattdelliott commented:
Maybe the better approach is to first check permissions on /Applications and if no write access then ~/Applications.

tommoor commented:
Right, so probably the order should just always be:

- Place in /Applications if user has write access
- Place in ~/Applications if it exists
- Fallback to requesting sudo for /Applications

mattdelliott commented:
Agreed. I can make those changes tomorrow AM.

mattdelliott committed:
Updated install location logic.

- Prefers root if it is writable
- If root is not writable and user applications exists, it will use that
- else it will ask for permission to write to root

mattdelliott committed:
Updated preferredInstallLocation to check ~/Applications permissions
